apps/evm/lib/evm/gas.ex

```diff
@@ -351,28 +351,44 @@ defmodule EVM.Gas do
     iex> exec_env = %EVM.ExecEnv{address: address, account_interface: account_interface}
     iex> EVM.Gas.operation_cost(:sstore, [0, 0], %EVM.MachineState{}, exec_env)
     5000
-    iex> EVM.Gas.operation_cost(:sstore, [0, 2], %EVM.MachineState{}, exec_env)
-    20000
   """
-  def operation_cost(:sstore, [key, new_value], _machine_state, exec_env) do
-    old_value = ExecEnv.get_storage(exec_env, key)
-
-    if old_value == 0 && new_value != 0 do
-      @g_sset
-    else
-      @g_sreset
+  def operation_cost(:sstore, params = [key, new_value], _machine_state, exec_env) do
+    case ExecEnv.get_storage(exec_env, key) do
+      :account_not_found ->
+        @g_sreset
+
+      :key_not_found ->
+        if new_value != 0 do
+          @g_sset
+        else
+          @g_sreset
+        end
+
+      {:ok, value} ->
+        if new_value != 0 && new_value != value && value == 0 do
+          @g_sset
+        else
+          @g_sreset
+        end
     end
   end
 
   def operation_cost(
         :call,
         [gas_limit, to_address, value, _in_offset, _in_length, _out_offset, _out_length],
-        _machine_state,
+        machine_state,
         exec_env
       ) do
     to_address = Address.new(to_address)
 
-    @g_call + call_value_cost(value) + new_account_cost(exec_env, to_address) + gas_limit
+    gas_limit = if exec_env.stack_depth == 1024, do: 0, else: gas_limit
+
+    call_value =
+      if exec_env.stack_depth == 1024 && value != 0,
```

Comment:
`max_stack_depth` should probably go in a config file at some point. Maybe just put it in a module var for now?

Reply:
@masonforest actually there is already variable in `Functions` module. Thanks!